validphys2/src/validphys/config.py

```diff
@@ -807,16 +807,29 @@ def produce_dataset_inputs_sampling_covmat(
         """
         from validphys import covmats
 
-        if use_thcovmat_in_sampling:
-            if sep_mult:
-                return covmats.dataset_inputs_total_covmat_separate
+        if use_t0_sampling:
+            if use_thcovmat_in_sampling:
+                if sep_mult:
+                    return covmats.dataset_inputs_t0_total_covmat_separate
+                else:
+                    return covmats.dataset_inputs_t0_total_covmat
             else:
-                return covmats.dataset_inputs_total_covmat
+                if sep_mult:
+                    return covmats.dataset_inputs_t0_exp_covmat_separate
+                else:
+                    return covmats.dataset_inputs_t0_exp_covmat
+
         else:
-            if sep_mult:
-                return covmats.dataset_inputs_exp_covmat_separate
+            if use_thcovmat_in_sampling:
+                if sep_mult:
+                    return covmats.dataset_inputs_total_covmat_separate
+                else:
+                    return covmats.dataset_inputs_total_covmat
             else:
-                return covmats.dataset_inputs_exp_covmat
+                if sep_mult:
+                    return covmats.dataset_inputs_exp_covmat_separate
+                else:
+                    return covmats.dataset_inputs_exp_covmat
```

Comment:
@scarlehoff @RoyStegeman 
If we reach consensus on using t0-covmat in sampling I would remove this bit here

Reply:
I think we might want to keep it even if we end up defaulting to `true`

Reply:
I'm not so sure yet, on the one had I'd rather get rid of functionality that is unused. Otoh, if we remove it here we should probably also remove the corresponding functions and I'm not so sure that's a good idea